src/components/echo.rs

```diff
@@ -14,35 +14,48 @@ use crate::components::Dimensions;
 use crate::components::DrawMode;
 use crate::Component;
 use crate::Line;
-use crate::State;
 
 /// Component that repeats whatever lines are put into it.
 /// Mostly useful for testing purposes.
 #[derive(Debug)]
-pub struct Echo<Msg> {
+pub struct Echo {
     collapse: bool,
-    _state: PhantomData<Msg>,
+    indices: Option<(usize, usize)>,
+    _state: PhantomData<Vec<Line>>,
 }
 
-impl<Msg> Echo<Msg> {
+impl Echo {
     pub fn new(collapse: bool) -> Self {
         Self {
             collapse,
+            indices: None,
+            _state: PhantomData,
+        }
+    }
+
+    pub fn new_with_indices(collapse: bool, indices: (usize, usize)) -> Self {
+        Self {
+            collapse,
+            indices: Some(indices),
             _state: PhantomData,
         }
     }
 }
 
-impl<Msg: AsRef<Vec<Line>> + Send + 'static + Debug> Component for Echo<Msg> {
+impl Component<Vec<Line>> for Echo {
```

Comment:
Previously `Msg` was used to to let `Echo` "pick" lines out of the total set of lines passed into state. As we are no longer using `State` this is impossible now. For tests that require only a subset of the total lines to be echoed I created the `new_with_indices` constructor that accepts a `(usize, usize)` representing the indices of a range of lines.